### PR TITLE
Category Create (Admin) 49990

### PR DESCRIPTION
**Pull request by abdullahallmehedi**

### CODE REVIEW

**Comment by manazirahsan** [comment]
okey

**Comment by Rownaknira** [comment]
ok

**Comment by SSalekin** [comment]
ok

File: config/routes.rb [hunk 1]
```diff
@@ -1,59 +1,6 @@
...
-  #   resources :photos, concerns: :toggleable
-
-  # Example resource route within a namespace:
-  #   namespace :admin do
-  #     # Directs /admin/products/* to Admin::ProductsController
-  #     # (app/controllers/admin/products_controller.rb)
-  #     resources :products
-  #   end
-end
+  namespace :admin do
```
Comment (dieunb) [inline config/routes.rb]: admin root page?

File: app/controllers/admin/categories_controller.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+class Admin::CategoriesController < ApplicationController
+  
+  def new
+    @category = Category.new
```
Comment (dieunb) [inline app/controllers/admin/categories_controller.rb]: after implement `cancancan`, pls fix it.

File: app/controllers/admin/categories_controller.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  end
+
+  def create
+    @category = Category.new category_params
```
Comment (dieunb) [inline app/controllers/admin/categories_controller.rb]: after implement `cancancan`, pls fix it.

File: app/controllers/registrations_controller.rb [hunk 4]
```diff
@@ -1,4 +1,7 @@
 class RegistrationsController < Devise::RegistrationsController
+  skip_before_action :authenticate_user!
```
Comment (dieunb) [inline app/controllers/registrations_controller.rb]: no need, i think so.

File: app/controllers/registrations_controller.rb [hunk 5]
```diff
@@ -1,4 +1,7 @@
...
+  skip_before_action :load_and_authorize_resource
```
Comment (dieunb) [inline app/controllers/registrations_controller.rb]: no need, i think so.

File: app/models/category.rb [hunk 6]
```diff
@@ -1,3 +1,7 @@
 class Category < ActiveRecord::Base
+  mount_uploader :image, CategoryImageUploader
   has_many :books, dependent: :destroy
-end
+  validates_presence_of [:name, :image]
+  validates :avatar, file_size: { less_than: 3.megabytes },
```
Comment (dieunb) [inline app/models/category.rb]: convention

File: app/models/category.rb [hunk 7]
```diff
@@ -1,3 +1,7 @@
...
+    file_content_type: { allow: ["image/jpeg", "image/png"] }
```
Comment (dieunb) [inline app/models/category.rb]: convention

**Pull request merged**